### PR TITLE
feat(db): add ability to check account verification status

### LINKED ISSUE #111: add ability to talk to fxa-db
We need ability to talk to the fxa db to check user's verification status.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #111

### CODE REVIEW

File: lib/crypto/butil.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
```
Comment (vladikoff) [inline lib/crypto/butil.js]: I wonder if we should make this a module, it is the same as in fxa-auth-server

cc @rfk

File: config.js [hunk 2]
```diff
@@ -19,6 +19,19 @@ const conf = convict({
...
+  db: {
+    backend: {
+      default: 'httpdb',
+      env: 'DB_BACKEND'
+    }
+  },
+  httpdb: {
+    url: {
+      doc: 'database api url',
+      default: 'http://127.0.0.1:8000',
```
Comment (vladikoff) [inline config.js]: TODO: this might need to be changed

File: lib/crypto/butil.js (hunk 1 above)
Comment (rfk) [inline lib/crypto/butil.js]: lol, I was just privately thinking "I'm not thrilled about all this code duplication from the auth-server" :-)

File: package.json [hunk 3]
```diff
@@ -24,13 +24,17 @@
     "handlebars": "1.3.0",
     "i18n-abide": "0.0.23",
     "jed": "0.5.4",
+    "joi": "6.9.1",
```
Comment (vladikoff) [inline package.json]: TODO: remove

File: lib/crypto/butil.js (hunk 1 above)
Comment (vladikoff) [inline lib/crypto/butil.js]: @jrm says @seanmonstar also has a buffer util module, need to check that

Comment (seanmonstar) [inline lib/crypto/butil.js]: https://www.npmjs.com/package/buf

Allows me to do `db.getToken(buf(token))` that works with either a hex string of a buffer, and then I can `reply({ token: hex(token) })` for the reverse: give me a hex string if its a Buffer, but don't error if it's already a string.

File: lib/crypto/butil.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+var HEX = /^(?:[a-fA-F0-9]{2})+$/
+
+module.exports.buffersAreEqual = function buffersAreEqual(buffer1, buffer2) {
```
Comment (seanmonstar) [inline lib/crypto/butil.js]: PS: `Buffer.equals` exists starting with nodejs 0.12. `bufOne.equals(bufTwo)` https://nodejs.org/dist/latest-v4.x/docs/api/buffer.html#buffer_buf_equals_otherbuffer

File: test/local/db_tests.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        t.pass('Got the ping ok')
+      }, function (err) {
+        throw err
+      })
+  }
+)
+
+
+test(
+  'get email record',
```
Comment (vbudhram) [inline test/local/db_tests.js]: The `pool.js` file exposes alot of functionality. Is it worthwhile to test other HTTP Methods for email?

File: lib/crypto/butil.js (hunk 1 above)
Comment (vladikoff) [inline lib/crypto/butil.js]: Ok thanks, looks like the `bufferize` in fxa-auth-server works on Objects, so I'm gonna keep that one for now.

File: test/local/db_tests.js (hunk 5 above)
Comment (vladikoff) [inline test/local/db_tests.js]: :+1:

**Comment by vladikoff** [comment]
@vbudhram updated, could you do another r? when you have time

**Comment by vbudhram** [comment]
@vladikoff LGTM

**Pull request merged**